Dockerfile

```diff
@@ -79,37 +76,49 @@ RUN conan export conan/mqtt_cpp \
 FROM os AS release
 
 LABEL author="mtconnect" description="Docker image for the latest Production MTConnect C++ Agent"
+ARG BIN_DIR=/usr/bin
+ARG MTCONNECT_DIR=/etc/mtconnect
+
+ENV MTCONNECT_DIR=$MTCONNECT_DIR
 
 # install ruby for simulator
-RUN apt-get update && apt-get install -y ruby
+RUN apt-get update && apt-get install -y ruby zip
 
 # change to a new non-root user for better security.
 # this also adds the user to a group with the same name.
 # --create-home creates a home folder, ie /home/<username>
 RUN useradd --create-home agent
 USER agent
+WORKDIR /home/agent
 
 # install agent executable
-COPY --chown=agent:agent --from=build /root/agent/build/bin/agent /usr/local/bin/
+COPY --chown=agent:agent --from=build /root/agent/*.zip .
 
 # copy data to /etc/mtconnect
-COPY --chown=agent:agent --from=build /root/agent/schemas /etc/mtconnect/schemas
-COPY --chown=agent:agent --from=build /root/agent/simulator /etc/mtconnect/simulator
-COPY --chown=agent:agent --from=build /root/agent/styles /etc/mtconnect/styles
+RUN unzip *.zip
+
+ENV ZIP_DIR=/home/agent/agent-*-Linux
```

Comment:
This won’t work, as `ENV` (and `ARG`) does not evaluate globs. In other words, `ZIP_DIR` is defined here as a string literal.

I see these solutions:

1. get the version part (currently set to `2.2.0.4` from somewhere;
2. create an archive (a ZIP) without the version number in the folder name (e.g. `agent-Linux`);
3. remove the `ZIP_DIR` env var and use the path explicitly (in `RUN`, globs work as expected).

Reply:
I was not expecting the ENV to resolve the glob, was just doing it to reduce typing and make it easier to find. 

The issue is cpack creates the file name for dist. I can see about changing the pattern in cmake. It would need to be passed through conan to cmake. 

Btw. I expected the `sh` to expand the env var glob.

Reply:
Well, I tried to unquote `$ZIP_DIR` (previously, I used `"$ZIP_DIR"`), however, it has failed me too.

```bash
 => ERROR [linux/amd64 release 6/7] RUN cp /home/agent/agent-*-Linux/bin/* "/usr/bin" &&   && mkdir -p "/etc/mtconnect"   && chown -R agent:agent "/etc/mtconnect"                                                                      0.7s 
------                                                                                                                                                                                                                                       
 > [linux/amd64 release 6/7] RUN cp /home/agent/agent-*-Linux/bin/* "/usr/bin" &&   && mkdir -p "/etc/mtconnect"   && chown -R agent:agent "/etc/mtconnect":                                                                                 
0.575 /bin/sh: 1: Syntax error: "&&" unexpected                                                                                                                                                                                              
------                                                                                                                                                                                                                                       
WARNING: No output specified with docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load                                           
Dockerfile:127                                                                                                                                                                                                                               
--------------------                                                                                                                                                                                                                         
 126 |     # RUN cp /home/agent/agent-*-Linux/bin/* "$BIN_DIR" \                                                                                                                                                                             
 127 | >>> RUN cp $ZIP_DIR/bin/* "$BIN_DIR" && \                                                                                                                                                                                             
 128 | >>>   && mkdir -p "$MTCONNECT_DIR" \                                                                                                                                                                                                  
 129 | >>>   && chown -R agent:agent "$MTCONNECT_DIR"                                                                                                                                                                                        
 130 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c cp $ZIP_DIR/bin/* \"$BIN_DIR\" &&   && mkdir -p \"$MTCONNECT_DIR\"   && chown -R agent:agent \"$MTCONNECT_DIR\"" did not complete successfully: exit code: 2
```

Reply:
I'm going to try something a little different when I run cpack create a tgz on *NIX platforms. I think I can change some of the directories where we can package with the destinations upfront and we just need to untar. Just testing now.

Reply:
Okay, but note that it might not matter if we use `tar xf` or `unzip`, you might know better.

Reply:
The difference is cpack can inject into tgz paths with a prefix if we know where the files should go. I need to tweak it a bit to figure out what is possible.